src/templates/Gallery.jsx

```diff
@@ -0,0 +1,244 @@
+    ) {
+      // continue on next page
+      setActiveIndex(0);
+      navigate(`${prefix}/${galleryCurrentPage + 1}?index=0`);
+    } else {
+      // continue on same page
+      setActiveIndex(activeIndex + 1);
+      navigate(`?index=${activeIndex + 1}`);
+    }
+  };
+
+  const showPrev = () => {
+    if (activeIndex <= 0 && galleryCurrentPage === 1) {
+      // continue on last page
+      setActiveIndex(limit - 1);
+      navigate(`${prefix}/${galleryNumPages}?index=${(numImages % limit) - 1}`);
+    } else if (activeIndex <= 0 && galleryCurrentPage !== 0) {
+      // continue on prev page
+      setActiveIndex(limit - 1);
+      navigate(
+        galleryCurrentPage === 2
+          ? `${prefix}?index=${limit - 1}`
+          : `${prefix}/${galleryCurrentPage - 1}?index=${limit - 1}`
+      );
+    } else {
+      // continue on same page
+      setActiveIndex(activeIndex - 1);
+      navigate(`?index=${activeIndex - 1}`);
+    }
+  };
+
+  const goToIndex = (newIndex) => {
+    setActiveIndex(newIndex);
+  };
+
+  const slides = imageList.map(({ id, name, date, image }) => (
+    <CarouselItem key={id}>
+      <GatsbyImage
+        image={image}
+        id="lightbox-img"
+        className="m-4"
+        imgStyle={{ boxShadow: "0 0 30px rgba(0, 0, 0, 0.5)" }}
+      />
+      <CarouselCaption
+        captionText={moment(date).format("MMMM DD, YYYY")}
+        captionHeader={name.replaceAll("_", " ")}
+      />
+    </CarouselItem>
+  ));
+
+  return (
+    <Layout title="Gallery">
+      {/* Modal Carousel */}
+      {imageDisplay ? (
+        <div className="media-modal">
+          <div className="lightbox">
+            <Col className="d-flex justify-content-end">
+              <button
+                type="button"
+                className="btn btn-lg btn-primary rounded-circle media-button-cancel"
+                onClick={hideImage}
+              >
+                <IconContext.Provider value={closeIconAttributes}>
+                  <ImCross />
+                </IconContext.Provider>
+              </button>
+            </Col>
+            <Col className="d-flex justify-content-center">
+              <Carousel
+                fade
+                slide={false}
+                activeIndex={activeIndex}
+                next={showNext}
+                prev={showPrev}
+              >
+                <CarouselIndicators
+                  items={imageList}
+                  activeIndex={activeIndex}
+                  onClickHandler={goToIndex}
+                />
+                {slides}
+                <CarouselControl
+                  direction="prev"
+                  directionText="<"
+                  onClickHandler={showPrev}
+                />
+                <CarouselControl
+                  direction="next"
+                  directionText=">"
+                  onClickHandler={showNext}
+                />
+              </Carousel>
+            </Col>
+          </div>
+        </div>
```

Comment:
Follow-up: Make this a component on it's own so that the overall layout of this page becomes easier to comprehend.